src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -316,6 +308,78 @@ public LockFileBuilder(int lockFileVersion, ILogger logger, Dictionary<RestoreTa
             return lockFile;
         }
 
+        private static void AddProjectFileDependenciesForSpec(PackageSpec project, LockFile lockFile)
+        {
+            // Use empty string as the key of dependencies shared by all frameworks
+            lockFile.ProjectFileDependencyGroups.Add(new ProjectFileDependencyGroup(
+                string.Empty,
+                project.Dependencies
+                    .Select(group => group.LibraryRange.ToLockFileDependencyGroupString())
+                    .OrderBy(group => group, StringComparer.Ordinal)));
+
+            foreach (var frameworkInfo in project.TargetFrameworks
+                .OrderBy(framework => framework.FrameworkName.ToString(),
+                    StringComparer.Ordinal))
+            {
+                lockFile.ProjectFileDependencyGroups.Add(new ProjectFileDependencyGroup(
+                    frameworkInfo.FrameworkName.ToString(),
+                    frameworkInfo.Dependencies
+                        .Select(x => x.LibraryRange.ToLockFileDependencyGroupString())
+                        .OrderBy(dependency => dependency, StringComparer.Ordinal)));
+            }
+        }
+
+        private static void AddProjectFileDependenciesForNETCore(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
```

Comment:
This was covered before for UWP and now for NETCore specifically in NETCoreProject2Project_ProjectFileDependencyGroupsForNETCore and also at the nuget.exe level.

The change is pretty small despite how it looks, the old behavior was to write out all dependencies from project.json. For NETCore we do that plus add project references. The code here is to find the graph entry for the project itself and copy the project references out. When the lock file is written out the project's graph entry is dropped otherwise consumers could just read it themselves.

UWP would probably be fine if projects were written out there also, I don't think anyone is consuming it. But just to be safe I check the project type and only write it for .NETCore.